src/lib.rs

```diff
@@ -1502,10 +1510,61 @@ impl rustc_serialize::Decodable for Url {
     }
 }
 
+#[cfg(feature="serde")]
+impl Url {
+    /// Serialize the URL efficiently, for use with IPC
+    pub fn serialize_efficient<S>(&self, serializer: &mut S) -> Result<(), S::Error> where S: serde::Serializer {
```

Comment:
Please don't name if `efficient`. Its main property is that it is risky, not efficient.